libraries/designsystem/src/main/kotlin/io/element/android/libraries/designsystem/components/ClickableLinkText.kt

```diff
@@ -143,10 +143,19 @@ fun AnnotatedString.linkify(linkStyle: SpanStyle): AnnotatedString {
         append(original)
         for (span in spans) {
             val start = spannable.getSpanStart(span)
-            val end = spannable.getSpanEnd(span)
+            var end = spannable.getSpanEnd(span)
             if (original.getLinkAnnotations(start, end).isEmpty() && original.getStringAnnotations("URL", start, end).isEmpty()) {
                 // Prevent linkifying domains in user or room handles (@user:domain.com, #room:domain.com)
                 if (start > 0 && !spannable[start - 1].isWhitespace()) continue
+
+                // Try to avoid including trailing punctuation in the link.
+                // Since this might fail in some edge cases, we catch the exception and just use the original end index.
+                runCatching {
+                    end = adjustEndIndex(spannable, start, end)
+                }.onFailure {
+                     Timber.e(it, "Failed to adjust end index for link span")
+                }
```

Comment:
This is adjusting the `end` of the span, so this is fixing the rendering issue, but it does this affect the URL itself, i.e. the target of the link.

I think the line `annotation = span.url,` below needs to be updated.

Reply:
It's maybe because it did not work with the example I tried:

<img width="254" alt="image" src="https://github.com/user-attachments/assets/6acd8552-428a-4f2b-becb-f56ebdb924b0" />

Reply:
Does the Event renderer use `ClickableLinkText`? It seems it's not the case 😅